src/yunohost/app.py

```diff
@@ -1659,6 +1721,129 @@ def _parse_app_instance_name(app_instance_name):
+
+        # Parse applist url in cron
+        cron_file_content = open(cron_path).read().strip()
+        applist_url_parse = re.search("-u (https?://[^ ]+)", cron_file_content)
+
+        # Abort if we did not find an url
+        if not applist_url_parse or not applist_url_parse.groups():
+            # Bkp the old cron job somewhere else
+            bkp_file = "/etc/yunohost/%s.oldlist.bkp" % applist_name
+            os.rename(cron_path, bkp_file)
+            # Notice the user
+            logger.warning(m18n.n('appslist_could_not_migrate',
+                           name=applist_name,
+                           bkp_file=bkp_file))
+        # Otherwise, register the list and remove the legacy cron
+        else:
+            applist_url = applist_url_parse.groups()[0]
+            try:
+                _register_new_applist(applist_url, applist_name)
+            # Might get an exception if two legacy cron jobs conflict
+            # in terms of url...
+            except Exception as e:
+                logger.error(str(e))
+                # Bkp the old cron job somewhere else
+                bkp_file = "/etc/yunohost/%s.oldlist.bkp" % applist_name
+                os.rename(cron_path, bkp_file)
+                # Notice the user
+                logger.warning(m18n.n('appslist_could_not_migrate',
+                               name=applist_name,
+                               bkp_file=bkp_file))
+            else:
+                os.remove(cron_path)
+
+    _install_applist_fetch_cron()
+
+
+def _install_applist_fetch_cron():
+
+    logger.debug("Installing applist fetch cron job")
+
+    with open("/etc/cron.daily/yunohost-fetch-applists", "w") as f:
+        f.write('#!/bin/bash\nyunohost app fetchlist > /dev/null 2>&1\n')
+
+
+def _read_applist_list():
+    """
+    Read the json corresponding to the list of applists
+    """
+
+    # If file does not exists yet, return empty dict
+    if not os.path.exists(APPLISTS_JSON):
+        return {}
+
+    # Read file content
+    with open(APPLISTS_JSON, "r") as f:
+        applists_json = f.read()
+
+    # Parse json, throw exception if what we got from file is not a valid json
+    try:
+        applists = json.loads(applists_json)
+    except ValueError:
+        raise MoulinetteError(errno.EBADR,
+                              m18n.n('appslist_corrupted_json', filename=APPLISTS_JSON))
+
+    return applists
+
+
+def _write_applist_list(applist_lists):
+    """
+    Update the json containing list of applists
+    """
+
+    # Read file content
+    with open(APPLISTS_JSON, "w") as f:
+        json.dump(applist_lists, f)
+
+
+def _register_new_applist(url, name):
+    """
+    Add a new applist to be fetched regularly.
+    Raise an exception if url or name conflicts with an existing list.
+    """
+
+    applist_list = _read_applist_list()
+
+    # Check if name conflicts with an existing list
+    if name in applist_list.keys():
```

Comment:
Ah indeed... Well, atm you can simply remove the list with removelist, then redo a fetchlist with the new url.